codice/ddf
# DDF-2162 Add the ability to inject attributes into metacard types

#### What does this PR do?
- Adds an API for registering 'injectable' attributes that should be injected into all metacard types or into specific metacard types
- Augments the JSON metacard type definition functionality by allowing attribute injections to be defined in JSON files
- Updates the GeoJSON and XML input transformers to inject the proper attributes on transformation so the injected attributes will be read from the GeoJSON and XML, respectively
- Updates the standard catalog framework implementation to inject the proper attributes on creates, updates, and queries

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@rzwiefel 
@bdeining
@pklinef  

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@jlcsmith
@kcwire 

#### How should this be tested?
Define some new attributes and metacard types using the JSON definition feature and make your new attributes injectable (following the JSON definition file documentation). Ingest some GeoJSON or XML metacards of various metacard types with values for the new attributes you defined. Verify that these attributes are present on each ingested metacard.

#### What are the relevant tickets?
[DDF-2162](https://codice.atlassian.net/browse/DDF-2162)

#### Checklist:

#### ==== Note ====
I still need to add unit tests for some of the modules.
